engine/src/main/java/org/terasology/rendering/dag/nodes/PrePostCompositeNode.java

```diff
@@ -58,23 +66,22 @@ public void initialise() {
     public void process() {
         PerformanceMonitor.startActivity("rendering/prePostComposite");
         prePostComposite.enable();
-        sceneOpaque = frameBuffersManager.getFBO("sceneOpaque");
-        sceneOpaquePingPong = frameBuffersManager.getFBO("sceneOpaquePingPong");
-        sceneReflectiveRefractive = frameBuffersManager.getFBO("sceneReflectiveRefractive");
+        sceneReflectiveRefractive = dynamicFBOsManager.get(REFLECTIVE_REFRACTIVE_URN);
 
         // TODO: verify if there should be bound textures here.
-        sceneOpaquePingPong.bind();
+        WRITE_ONLY_GBUFFER.bind();
 
-        setViewportToSizeOf(sceneOpaquePingPong);
+        setViewportToSizeOf(WRITE_ONLY_GBUFFER);
         glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT); // TODO: verify this is necessary
 
         renderFullscreenQuad();
 
         bindDisplay();     // TODO: verify this is necessary
-        setViewportToSizeOf(sceneOpaque);    // TODO: verify this is necessary
+        setViewportToSizeOf(READ_ONLY_GBUFFER); // TODO: verify this is necessary
+
+        dynamicFBOsManager.swapReadWriteBuffers();
 
-        frameBuffersManager.swapSceneOpaqueFBOs();
-        sceneOpaque.attachDepthBufferTo(sceneReflectiveRefractive);
+        READ_ONLY_GBUFFER.attachDepthBufferTo(sceneReflectiveRefractive);
```

Comment:
Resulting image is: (notice the right side of the final image)

![screen shot 2016-08-19 at 02 38 37](https://cloud.githubusercontent.com/assets/136392/17794231/456f542a-65b6-11e6-83bb-f2cc3f10ee66.png)